src/test/conditions/rsa_sha256_test.cpp

```diff
@@ -0,0 +1,312 @@
+        "KmbxNu7o+zMyOblNg/I6CQMSEuo6jHoLVc9QaODscGco3du8WjRwJ2DnA3HoBJ5F\n"
+        "L0XftzOCfSrBQfn0Fb2ej4nsaIw1z0wEaAnuDC18/VUHQ0rHl2K2QleX4FwBiyXK\n"
+        "qWzhAVuxskkfWe3Xgn58IT2MODSDnFhP8j6m0vq5lklwgfIi9c6+y0rmJbhSZI4N\n"
+        "b/o5HSpWAxfpaSnWzw5moN5JP6DmhGQzgnctW9YL2w4OfZ9jPHl+xWMlSGUd8TD2\n"
+        "QIpo2Qox/w==\n"
+        "-----END RSA PRIVATE KEY-----\n";
+
+    void check (
+        fulfillment_t const& f,
+        condition_t const& c,
+        Slice test,
+        Slice good)
+    {
+        BEAST_EXPECT (validate (f, c, test) ==
+            ((test == good) && (f.condition() == c)));
+
+        BEAST_EXPECT (validate (f, test) ==
+            (test == good));
+    }
+
+    void testKnown ()
+    {
+        testcase ("Known");
+
+        Slice m = makeSlice (knownMessage);
+        std::string test = "aaabc";
+
+        // Load and test string and binary and text
+        // serialization & deserialization
+        auto const f = load_fulfillment(knownFulfillment);
+        BEAST_EXPECT (f);
+        BEAST_EXPECT (to_string (*f) == knownFulfillment);
+
+        {
+            auto const f2 = load_fulfillment(makeSlice(to_blob(*f)));
+            BEAST_EXPECT (f2);
+            BEAST_EXPECT (*f == *f2);
+        }
+
+        // Verify the condition for this fulfillment and test
+        // binary and text serialization & deserialization
+        auto const c = f->condition();
+        BEAST_EXPECT (to_string(c) == knownCondition);
+
+        {
+            auto c1 = load_condition (knownCondition);
+            BEAST_EXPECT (c1);
+            BEAST_EXPECT (c == *c1);
+
+            auto c2 = load_condition (makeSlice(to_blob(c)));
+            BEAST_EXPECT (c2);
+            BEAST_EXPECT (c == *c2);
+        }
+
+        // First check against incorrect conditions, using both
+        // correct, incorrect and empty buffers.
+        char const* const ccs[] =
+        {
+            "cc:0:3:PWh2oBRt6FdusjlahY3hIT0bksZbd53zozHP1aRYRUY:256",
+            "cc:1:25:XkflBmyISKuevH8-850LuMrzN-HT1Ds9zKUEzaZ2Wk0:103",
+            "cc:2:2b:d3O4epRCo_3rj17Bf3v8hp5ig7vq84ivPok07T9Rdl0:146",
+            "cc:3:11:Mjmrcm06fOo-3WOEZu9YDSNfqmn0lj4iOsTVEurtCdI:518",
+            "cc:4:20:O2onvM62pC1io6jQKm8Nc2UyFXcd4kOmOsBIoYtZ2ik:96"
+        };
+
+        for (auto cc : ccs)
+        {
+            auto nc = load_condition (cc);
+
+            if (BEAST_EXPECT (nc && nc != c))
+            {
+                check (*f, nc.get(), makeSlice(test), m);
+                check (*f, nc.get(), m, m);
+                check (*f, nc.get(), { }, m);
+            }
+        }
+
+        // Now try against the correct condition with various
+        // buffers - most are incorrect, some are correct.
+        do
+        {
+            for (Slice t = makeSlice (test); !t.empty(); t += 1)
+                check (*f, c, t, m);
+        } while (std::next_permutation(test.begin(), test.end()));
```

Comment:
Very nice use of permutation. You even have repeated values in `test` so you get more correct conditions. I like it.